File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//定义服务层  与后端进行请求交互。
app.service("brandService",function ($http) {
    //查询所有
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    //分页查询
    this.findPage=function (pageNum,pageSize,searchEntity) {
        return $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+ pageSize,searchEntity);
    }

    //新增
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    //修改
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }
    //根据id查询
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //批量删除
    this.dele=function (ids) {
        return  $http.get("../brand/delete.do?ids="+ids);
    }

    this.selectBrandList=function () {
        return $http.get("../brand/selectBrandList.do");
    }
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 查询所有
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询品牌数据
     */
    public PageResult findPage(Integer pageNum,Integer pageSize);

    void add(TbBrand brand);

    TbBrand findOne(Long id);

    void update(TbBrand brand);

    void delete(Long[] ids);

    PageResult findPage(TbBrand brand, int page, int rows);

    List<Map> selectBrandList();
}
